Remora.Discord.Commands/Attributes/DiscordDefaultDMPermissionAttribute.cs

```diff
@@ -0,0 +1,37 @@
+//
+//  DiscordDefaultDMPermissionAttribute.cs
+//
+//  Author:
+//       Jarl Gullberg <jarl.gullberg@gmail.com>
+//
+//  Copyright (c) 2017 Jarl Gullberg
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+using System;
+
+namespace Remora.Discord.Commands.Attributes;
+
+/// <summary>
+/// Marks a command group as being either executable or non-executable in a DM.
+/// </summary>
+[AttributeUsage(AttributeTargets.Class, AllowMultiple = false)]
```

Comment:
Re: a remark in the server earlier; this should probably be allowed for methods as well, in the case of top-level commands in an unnamed group.

Reply:
Hmm, doesn't this also need special detection when registering?

Reply:
Most likely yes, you'd have to implement that too (+tests).